automation/jenkins_generate_ami.sh

```diff
@@ -42,12 +43,12 @@ YOCTO_IMAGES_PATH="${WORKSPACE}/build/tmp/deploy/images/${MACHINE}"
 IMAGE_NAME=${IMAGE_NAME:-balena-image-${MACHINE}.balenaos-img}
 
 ORIG_IMAGE="${YOCTO_IMAGES_PATH}/${IMAGE_NAME}"
-PRELOADED_IMAGE="$(mktemp -p ${YOCTO_IMAGES_PATH})"
+PRELOADED_IMAGE="$(mktemp -p "${YOCTO_IMAGES_PATH}")"
 
 cp "${ORIG_IMAGE}" "${PRELOADED_IMAGE}"
 
 # AMI names must be between 3 and 128 characters long, and may contain letters, numbers, '(', ')', '.', '-', '/' and '_'
-VERSION=$(cat "${YOCTO_IMAGES_PATH}/VERSION_HOSTOS" | sed 's/+/-/g')
+VERSION=$(cat < "${YOCTO_IMAGES_PATH}/VERSION_HOSTOS" | sed 's/+/-/g')
```

Comment:
Why the extra redirect here?

Reply:
https://github.com/koalaman/shellcheck/wiki/SC2002

```
In automation/jenkins_generate_ami.sh line 50:                                                                                                        
VERSION=$(cat "${YOCTO_IMAGES_PATH}/VERSION_HOSTOS" | sed 's/+/-/g')                                                                                  
              ^-- SC2002 (style): Useless cat. Consider 'cmd < file | ..' or 'cmd file | ..' instead.                                                 
```

Reply:
I think the proper way to address that warning is to remove the cat altogether.
```suggestion
VERSION="$(sed 's/+/-/g' "${YOCTO_IMAGES_PATH}/VERSION_HOSTOS")"
```

Reply:
both are valid, but yes, the later is more concise.